aiida/cmdline/utils/migration/v03_to_v04.py

```diff
@@ -0,0 +1,452 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Migration from v0.3 to v0.4, used by `verdi export migrate` command."""
+# pylint: disable=too-many-locals,too-many-statements,too-many-branches,unused-argument
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+
+import numpy as np
+
+from aiida.cmdline.utils import echo
+from aiida.cmdline.utils.migration.utils import verify_metadata_version, update_metadata
+
+__all__ = ('migrate_v3_to_v4',)
+
+
+def migrate_provenance_redesign(data):
+    """Utility function for `migrate_v3_to_v4` to handle REV. 1.0.20 (provenance redesign)"""
+    # step1: rename the type column of process nodes
```

Comment:
perhaps move this down - currently it seems like one can return after step1 before getting to step2

Reply:
Done